Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -17,152 +19,89 @@ class BTAnalyticsService: Equatable {
         }
     }
     
     /// Exposed to be able to execute this function synchronously in unit tests
-    func performEventRequest(
-        _ eventName: String,
-        connectionStartTime: Int? = nil,
-        correlationID: String? = nil,
-        endpoint: String? = nil,
-        endTime: Int? = nil,
-        errorDescription: String? = nil,
-        isVaultRequest: Bool? = nil,
-        linkType: String? = nil,
-        payPalContextID: String? = nil,
-        requestStartTime: Int? = nil,
-        startTime: Int? = nil
-    ) async {
-        let timestampInMilliseconds = Date().utcTimestampMilliseconds
-        let event = FPTIBatchData.Event(
-            connectionStartTime: connectionStartTime,
-            correlationID: correlationID,
-            endpoint: endpoint,
-            endTime: endTime,
-            errorDescription: errorDescription,
-            eventName: eventName,
-            isVaultRequest: isVaultRequest,
-            linkType: linkType,
-            payPalContextID: payPalContextID,
-            requestStartTime: requestStartTime,
-            startTime: startTime,
-            timestamp: String(timestampInMilliseconds)
-        )
-
-        await BTAnalyticsService.events.append(event)
+    func performEventRequest(with event: FPTIBatchData.Event) async {
+        await events.append(event)
         
         if shouldBypassTimerQueue {
             await self.sendQueuedAnalyticsEvents()
         }
     }
 
-    // MARK: - Helpers
+    // MARK: - Private Methods
 
-    func sendQueuedAnalyticsEvents() async {
-        if await !BTAnalyticsService.events.isEmpty {
+    private func sendQueuedAnalyticsEvents() async {
+        if await !events.isEmpty, let apiClient {
             do {
                 let configuration = try await apiClient.fetchConfiguration()
                 let postParameters = await createAnalyticsEvent(
                     config: configuration,
                     sessionID: apiClient.metadata.sessionID,
-                    events: Self.events.allValues
+                    events: events.allValues
                 )
                 http?.post("v1/tracking/batch/events", parameters: postParameters) { _, _, _ in }
-                await Self.events.removeAll()
+                await events.removeAll()
             } catch {
                 return
             }
         }
     }
 
     /// Constructs POST params to be sent to FPTI
-    func createAnalyticsEvent(config: BTConfiguration, sessionID: String, events: [FPTIBatchData.Event]) -> Codable {
+    private func createAnalyticsEvent(config: BTConfiguration, sessionID: String, events: [FPTIBatchData.Event]) -> Codable {
         let batchMetadata = FPTIBatchData.Metadata(
-            authorizationFingerprint: apiClient.authorization.type == .clientToken ? apiClient.authorization.bearer : nil,
+            authorizationFingerprint: apiClient?.authorization.type == .clientToken ? apiClient?.authorization.bearer : nil,
```

Comment:
I think `metadata` and `authorization` are constants in BTAPIClient (they are var for testing purposes).  In the future, we might consider removing the `BTAPIClient` dependency from here altogether and instead inject `metadata`, `authorization`, and `Configuration`

Reply:
Yeh this is a good callout. Though it did cross my mind going back to this PR (https://github.com/braintree/braintree_ios/pull/1337) that we don't need a config to be present for analytics at the time of `BTAPIClient.sendAnalytics()` but rather right before we flush/batch upload the analytics after the 15 seconds timer.

If we inject the Config from the BTAPIClient, we might get back to calling the `fetchOrReturnRemoteConfig` more times than necessary